include/graphblas/algorithms/amg/amg_data.hpp

```diff
@@ -0,0 +1,201 @@
+		 *        note that this information is automatically
+		 *        destructed on object destruction (if any)
+		 *
+		 * Vectors stored here refer to the \b coarsened system
+		 * (with the exception of #Ax_finer),
+		 * thus having size #system_size; this also holds for the system matrix #A,
+		 * while #coarsening_matrix has size #system_size \f$ \times \f$ #finer_size.
+		 * Hence, the typical usage of this data structure is to coarsen \b external
+		 * vectors, e.g. vectors coming from another
+		 * \code multi_grid_data<IOType, NonZeroType>
+		 * \endcode object whose #system_size equals
+		 * \code this-> \endcode #fines_size, via
+		 * \code this-> \endcode #coarsening_matrix and store the coarsened
+		 * vectors internally. Mimicing the recursive behavior of standard multi-grid
+		 * simulations, the information for a further coarsening is stored inside
+		 * #coarser_level, so that the  hierarchy of coarsened levels is reflected
+		 * inside this data structure.
+		 *
+		 * As for \ref system_data, internal vectors and matrices are initialized to the
+		 * proper size, but their values are \b not initialized.
+		 */
+		template< typename IOType, typename NonzeroType >
+		struct multi_grid_data : public system_data< IOType, NonzeroType > {
+
+			const std::size_t finer_size; ///< ssize of the finer system to coarse from;
+			grb::Vector< IOType > Ax_finer; ///< finer vector for intermediate computations, of size #finer_size
+			grb::Matrix< NonzeroType > coarsening_matrix; ///< matrix of size #system_size \f$ \times \f$ #finer_size
+			                                              ///< to coarsen an input vector of size #finer_size
+			                                              ///< into a vector of size #system_size
+			struct multi_grid_data< IOType, NonzeroType > *coarser_level; ///< pointer to next coarsening level, for recursive
+			                                                               ///< multi-grid V cycle implementations
+
+			/**
+			 * Construct a new \c multi_grid_data_object by initializing internal
+			 *     data structures and setting #coarser_level to \c nullptr.
+			 * @param[in] coarser_size size of the current system,
+			 *            i.e. size \b after coarsening
+			 * @param[in] _finer_size  size of the finer system,
+			 *            i.e. size of external objects \b before coarsening
+			 */
+			multi_grid_data( std::size_t coarser_size, std::size_t _finer_size ) :
+				system_data< IOType, NonzeroType >( coarser_size ), finer_size( _finer_size ),
+				Ax_finer( finer_size ), coarsening_matrix( coarser_size, finer_size ) {
+				coarser_level = nullptr;
+			}
+
+			/**
+			 * @brief Destroys the \c multi_grid_data_object object
+			 * by destroying #coarser_level.
+			 */
+			virtual ~multi_grid_data() {
+				if( coarser_level != nullptr ) {
+					delete coarser_level;
+				}
+			}
+		};
+
+		/**
+		 * @brief Data stucture to store the data for a full AMG run:
```

Comment:
Please remove `@brief` (the first line is always the brief description) and use a dot instead of `:` (otherwise it's not brief:)